app/models/team.rb

```diff
@@ -0,0 +1,82 @@
+class Team < ActiveRecord::Base
+  attr_reader :employee_tokens, :team_lead_id
+  belongs_to :department
+  belongs_to :company
+  has_many :employee_teams
+  has_many :employees, through: :employee_teams, dependent: :destroy
+  validates :name, :department_id, presence: true
+
+  def create_team(team_lead_id, employee_ids)
+    team_members_ids = employee_ids(employee_ids)
+    begin
+      self.transaction do
+        self.save!
+        self.employee_teams.create!(employee_id: team_lead_id, employee_type: 'Team Leader')
+        if team_members_ids.empty?
+          self.errors.add(:base, 'Team members name required')
+          return false
+        end
+        team_members_ids.each do |team_member|
+          self.employee_teams.create!(employee_id: team_member, employee_type: 'Team Member')
+        end
+        return self
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      self.add_errors(e)
+      false
+    rescue Exception
+      false
+    end
+  end
+
+  def update_team(team_lead_id, employee_ids)
+    team_members_ids = employee_ids(employee_ids)
+    begin
+      self.transaction do
+        self.save!
+        team_employees = self.employee_teams
```

Comment:
probably the lines where we're getting/finding from database can be moved out of the transaction?